pkg/apis/cluster/types.go

```diff
@@ -117,9 +117,18 @@ type ClusterSpec struct {
 	Region string
 
 	// Zone represents the zone of the member cluster locate in.
+	// Deprecated: This filed was never been used, use Zones instead.
 	// +optional
 	Zone string
```

Comment:
In our environment, this field is already in use, when we upgrade Karmada, the value of the `zone` field will not be populated to zones, so the `zones` field is still empty. I wonder if it is necessary to autofill the value of `zone` to `zones` in next pr. If so, we'll only care about `zones` and it's ok even if the `zone` field is removed in the future.

Reply:
> Disable the deprecated zone in v1.8 by introducing validation

except for this point, I think others make sense. can we disable the deprecated `zone` field in v1.9 or later to allow more time for users to adapt? we won't upgrade Karmada to v1.7 because our upgrade cycle is usually half a year. so we will choose to upgrade directly to v1.8.

Reply:
Sure. Then, we won't introduce the validation.
I will add more comments on the deprecated field to note that the field will never be removed from `v1alpha1`.

Reply:
Hi @whitewindmills Please take another look, and I just opened an issue (#3952) to track subsequent tasks. Some of them might be a little bit challenge.

Reply:
> and I just opened an issue (https://github.com/karmada-io/karmada/issues/3952) to track subsequent tasks. Some of them might be a little bit challenge.

thanks, I will try some tasks.